src/Notify/Client/NotificationClient.cs

```diff
@@ -4,190 +4,204 @@
-        public String SEND_EMAIL_NOTIFICATION_URL = "v2/notifications/email";
-        public String SEND_LETTER_NOTIFICATION_URL = "v2/notifications/letter";
-        public String GET_TEMPLATE_URL = "v2/template/";
-        public String GET_ALL_TEMPLATES_URL = "v2/templates";
-        public String TYPE_PARAM = "?type=";
-        public String VERSION_PARAM = "/version/";
-
-        public NotificationClient(String apiKey) : base(new HttpClientWrapper(new HttpClient()), apiKey)
+        public string GET_RECEIVED_TEXTS_URL = "v2/received-text-messages";
+        public string GET_NOTIFICATION_URL = "v2/notifications/";
+        public string SEND_SMS_NOTIFICATION_URL = "v2/notifications/sms";
+        public string SEND_EMAIL_NOTIFICATION_URL = "v2/notifications/email";
+        public string SEND_LETTER_NOTIFICATION_URL = "v2/notifications/letter";
+        public string GET_TEMPLATE_URL = "v2/template/";
+        public string GET_ALL_NOTIFICATIONS_URL = "v2/notifications";
+        public string GET_ALL_TEMPLATES_URL = "v2/templates";
+        public string TYPE_PARAM = "?type=";
+        public string VERSION_PARAM = "/version/";
+
+        public NotificationClient(string apiKey) : base(new HttpClientWrapper(new HttpClient()), apiKey)
         {
-
         }
 
-        public NotificationClient(String baseUrl, String apiKey) : base(new HttpClientWrapper(new HttpClient()), apiKey, baseUrl)
+        public NotificationClient(string baseUrl, string apiKey) : base(new HttpClientWrapper(new HttpClient()), apiKey,
+            baseUrl)
         {
-
         }
 
-        public NotificationClient(IHttpClient client, String apiKey) : base(client, apiKey)
+        public NotificationClient(IHttpClient client, string apiKey) : base(client, apiKey)
         {
-
         }
 
-        public Notification GetNotificationById(String notificationId)
+        public Notification GetNotificationById(string notificationId)
         {
-            String url = GET_NOTIFICATION_URL + notificationId;
+            var url = GET_NOTIFICATION_URL + notificationId;
 
-            String response = this.GET(url);
+            var response = this.GET(url);
 
             try
             {
-                Notification notification = JsonConvert.DeserializeObject<Notification>(response);
+                var notification = JsonConvert.DeserializeObject<Notification>(response);
                 return notification;
             }
             catch (JsonReaderException)
             {
                 throw new NotifyClientException("Could not create Notification object from response: {0}", response);
             }
-
         }
 
-        public NotificationList GetNotifications(String templateType = "", String status = "", String reference = "", String olderThanId = "")
+        public static string ToQueryString(NameValueCollection nvc)
         {
-            var query = HttpUtility.ParseQueryString("");
+            if (nvc.Count == 0) return "";
```

Comment:
Not sure what the gds coding standards say - but generally this one liner isn't consistent with the rest of the file

if (nvc.Count == 0) 
{
    return "";
}